src/pages/CreatePool/index.tsx

```diff
@@ -47,8 +52,23 @@ export type ValidationStatus =
 export default function CreatePool(): React.ReactElement {
   const { t } = useTranslation()
   const theme = useTheme()
-  const poolRegistry = usePoolRegistry()
-  const { account, library, chainId } = useActiveWeb3React()
+  const expandedPools = useContext(ExpandedPoolsContext)
+  const expandedPoolsRemapped = Object.values(expandedPools.data.byName)
+    .map((pool) => ({
```

Comment:
Could we reverse the order of `.filter` and `.map` and only run `.filter` once? Running `.filter` first can lead to `.map` iterating over less items.

We can combined those filter conditions into a single Boolean. Reduction from O(5n) to O(n) could be noticeable. 

Could be something like this. Up to your discretion to use it verbatim or not.  cc: @hammeiam. Want to see your thoughts on this.

```
const isPoolValid = (pool) => {
  return (
    pool.tokensLength < 4 &&
    pool.isSaddleApproved &&
    !pool.basePoolAddress &&
    !pool.isMigrated &&
    !pool.isPaused
  )
}
const expandedPoolsRemapped = Object.values(expandedPools.data.byName)
  .filter(isPoolValid)
  .map((pool) => ({
    basePoolAddress: pool.basePoolAddress,
    poolName: pool.poolName,
    isSaddleApproved: pool.isSaddleApproved,
    address: pool.poolAddress,
    isPaused: pool.isPaused,
    isMigrated: pool.isMigrated,
    tokensLength: pool.tokens.length,
  }))